modules/rostests/rosautotest/main.cpp

```diff
@@ -77,8 +77,27 @@ wmain(int argc, wchar_t* argv[])
                       NULL,
                       NULL);
 
+        if (Configuration.GetRepeatCount() > 1)
+        {
+            stringstream ss1;
+
+            ss1 << "[ROSAUTOTEST] The test will be repeated " << Configuration.GetRepeatCount() << " times" << endl;
+            StringOut(ss1.str());
+        }
+
         /* Run the tests */
-        WineTest.Run();
+        for (int i = 0; i < Configuration.GetRepeatCount(); i++)
+        {
+            CWineTest WineTest;
+
+            if (Configuration.GetRepeatCount() > 1)
+            {
+                stringstream ss;
+                ss << "[ROSAUTOTEST] Running attempt #" << i+1 << endl;
+                StringOut(ss.str());
+            }
+            WineTest.Run();
```

Comment:
Strange that this function is used like that. I would have expected it's instead some kind of static function of the CWineTest class ^^ (to be used like `CWineTest::Run()`).

Reply:
then how is it supposed to access member variables?

Reply:
besides, Run is virtual...